NumberTheory/PerfectNumbers.cpp

```diff
@@ -0,0 +1,37 @@
+ /*
+    Perfect number:
+    prints up to N perfect numbers (a positive integer that 
+    is equal to the sum of its proper positive divisors)
+    Time complexity:
+    O(sqrt(N)), where N is the number being checked 
+    Space complexity:
+    O(1), constant amount of space needed
+*/
+
+#include <iostream>
+
+using namespace std;
+
+bool is_perfect(int num) {
+	int sum = 1;
+	for(int divisor = 2; divisor <= num/2; divisor++)

```

Comment:
`num`, `sum` and `divisor` should be of type `unsigned long long` so that they can go upto 2<sup>64</sup>-1.